Cabal/src/Distribution/Simple/Build/PathsModule/Z.hs

```diff
@@ -41,6 +41,14 @@ render z_root = execWriter $ do
     return ()
   else do
     return ()
+  if (zSupportsCpp z_root)
+  then do
+    tell "#if __GLASGOW_HASKELL__ >= 810\n"
+    tell "{-# OPTIONS_GHC -fno-warn-prepositive-qualified-module #-}\n"
```

Comment:
Since this is for GHC >= 8 only, you could use the new syntax `-Wno-prepositive-qualified-module`.